homeassistant/components/conversation/default_agent.py

```diff
@@ -214,10 +214,15 @@ def _listen_clear_slot_list(self) -> None:
         ]
 
     async def async_recognize(
-        self, user_input: ConversationInput
+        self,
+        user_input: ConversationInput,
+        strict_intents_only: bool = False,
+        match_sentence_triggers: bool = True,
     ) -> RecognizeResult | SentenceTriggerResult | None:
         """Recognize intent from user input."""
-        if trigger_result := await self._match_triggers(user_input.text):
+        if (match_sentence_triggers) and (
+            trigger_result := await self._match_triggers(user_input.text)
+        ):
```

Comment:
Remove this feature of `async_recognize` and call it manually before calling `async_recognize`. Now there is 1 call that wants it, 1 that doesn't.

Reply:
I've split `async_recognize` into `async_recognize_sentence_trigger` and `async_recognize_intent`.